src/category_theory/category/Bipointed.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import category_theory.concrete_category.basic
+
+/-!
+# The category of bipointed types
+
+This defines `Bipointed` the category of bipointed types.
+
+## TODO
+
+Monoidal structure
+-/
+
+open category_theory
+
+universes u
+variables {α β : Type*}
+
+/-- The category of bipointed types. -/
+structure Bipointed : Type.{u + 1} :=
+(X : Type.{u})
+(to_prod : X × X)
+
+namespace Bipointed
+
+instance : has_coe_to_sort Bipointed Type* := ⟨Bipointed.X⟩
+
+/-- Turns a bipointing into a bipointed type. -/
+def of {X : Type*} (to_prod : X × X) : Bipointed := ⟨X, to_prod⟩
+
+alias of ← prod.Bipointed
+
+instance : inhabited Bipointed := ⟨of ((), ())⟩
+
+/-- Morphisms in `Bipointed`. -/
+@[nolint has_inhabited_instance, ext]
```

Comment:
I personally like this linter, but I know opinions vary! In this case though, I think you can just satisfy it by using `id` below?

Reply:
Got that from `simplex_category`, but I agree!